src/types/redis_string.cc

```diff
@@ -373,22 +373,20 @@ rocksdb::Status String::MSet(const std::vector<StringPair> &pairs, uint64_t ttl)
   // Data race, key string maybe overwrite by other key while didn't lock the key here,
   // to improve the set performance
   std::string ns_key;
+  auto batch = storage_->GetWriteBatchBase();
+  WriteBatchLogData log_data(kRedisString);
+  batch->PutLogData(log_data.Encode());
   for (const auto &pair : pairs) {
     std::string bytes;
     Metadata metadata(kRedisString, false);
     metadata.expire = expire;
     metadata.Encode(&bytes);
     bytes.append(pair.value.data(), pair.value.size());
-    auto batch = storage_->GetWriteBatchBase();
-    WriteBatchLogData log_data(kRedisString);
-    batch->PutLogData(log_data.Encode());
     AppendNamespacePrefix(pair.key, &ns_key);
     batch->Put(metadata_cf_handle_, ns_key, bytes);
     LockGuard guard(storage_->GetLockManager(), ns_key);
```

Comment:
what does this guard for?

Reply:
It uses to tell tools like `kvrocks2redis` about how to process the current batch.

Reply:
Previously, the code is

```
LockGuard guard(storage_->GetLockManager(), ns_key);
auto s = storage_->Write(storage_->DefaultWriteOptions(), batch->GetWriteBatch());
if (!s.ok()) return s;
```

but now `storage_->Write` is removed, so I wonder if LockGuard is a nop?

Reply:
I misunderstood your point. We indeed can remove this guard since the RocksDB is thread-safe for the single batch.

Reply:
So lets wait for another patch merged first